js/scripts.js

```diff
@@ -190,40 +171,108 @@ let pokemonRepository = (function() {
         button.classList.add(
             'pokemon-list__pokemon-card',
             'btn',
+            'shadow',
             'col-12',
             'col-md-4',
             'col-lg-3',
-            pokemon.mainType
+            `id-${pokemon.id}`,
+            `gradient--${pokemon.mainType}`
             );
+        button.setAttribute('id', pokemon.name);
         button.setAttribute('data-toggle', 'modal');
         button.setAttribute('data-target', '#pokemodal');
 
+        // Create flex element to hold name and id elements
+        let buttonTextWrapper = document.createElement('div');
+        buttonTextWrapper.classList.add('pokemon-list__pokemon-card--text');
+
+        // Create name element to include in button
+        let buttonName = document.createElement('p');
+        buttonName.innerText = `${pokemon.name}`;
+
+        // Create image element to include in button
+        let buttonId = document.createElement('p');
+        buttonId.innerText = `#${pokemon.id}`;
+
+        // Create image element to include in button
+        let buttonImage = document.createElement('img');
+        buttonImage.setAttribute('src', pokemon.imageUrl);
+
         // Add click event to button to show item details.
         addEvent(button, pokemon);
 
+        buttonTextWrapper.appendChild(buttonName);
+        buttonTextWrapper.appendChild(buttonId);
+        button.appendChild(buttonTextWrapper);
+        button.appendChild(buttonImage);
         htmlList.appendChild(button);
     }
 
+  // Search through pokémon by item name
+  let searchValue = document.getElementById('searchBar');
+
+  searchValue.addEventListener('keyup', function (e) {
+    let searchString = e.target.value.toLowerCase();
+
+    let itemsToHide = pokemonList.filter(function (item) {
+      // Find non-matching names
+      if (
+        !item.name.toLowerCase().includes(searchString)
+      ) {
+        return item;
+      }
+    });
+    let itemsToShow = pokemonList.filter(function (item) {
+      // Find matching names
+      if (
+        item.name.toLowerCase().includes(searchString)
+      ) {
+        return item;
+      }
+    });
+
+    itemsToHide.map((item) => {
+      document.getElementById(item.name).classList.add('d-none');
+    });
+    itemsToShow.map((item) => {
+      document.getElementById(item.name).classList.remove('d-none');
+    });
+  });
+
+    // Toggle search bar by clicking the search icon
+    let searchButton = document.querySelector('.search-button');
+    $(document).ready(function () {
+        $(searchButton).click(function () {
+            $('.search-bar').toggleClass('d-none');
+        });
+    });
+
     return {
         addListItem,
         loadList,
-        getAll
+        getAll,
+        loadDetails,
+        showLoadingMessage,
+        hideLoadingMessage
     };
 })();
 
-// Writes the names of each pokémon to the DOM
+// Load all the data and create the card list
+pokemonRepository.showLoadingMessage();
```

Comment:
IMO the `showLoadingMessage` and `hideLoadingMessage` functions should be private. WDYT?